### PR TITLE
#21896 recognize precision for bigint attributes

### LINKED ISSUE #21896: Snowflake numeric columns always show as number(38) in explorer
### Description

In the Dbeaver Community edition database navigator, numeric columns always show as NUMBER(38), regardless of their actual datatype (screenshot attached). Other datatypes seem to be correct.

Dbeaver snowflake issue

### DBeaver Version

Community Edition 23.2.5

### Operating System

Windows 10

### Database and driver

Snowflake, using snowflake-jdbc-3.14.3.jar

### Steps to reproduce

Connect to snowflake. Browse any Snowflake table with a numeric column that is NOT number(38), and it shows in Dbeaver as number(38).

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by LonwoLonwo**
2023-11-20 12_06_52-DBeaver Ultimate 23 2 5 - EMPLOYEES

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/TableCache.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ protected GenericTableColumn fetchChild(@NotNull JDBCSession session, @NotNull G
             log.warn("Error getting column scale", e);
         }
         Integer precision = null;
-        if (valueType == Types.NUMERIC || valueType == Types.DECIMAL) {
+        if (valueType == Types.NUMERIC || valueType == Types.DECIMAL || valueType == Types.BIGINT) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/TableCache.java]: Not sure that we should use for all drivers.
BIGINT doesn't have precision (it is an integer).

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/TableCache.java (hunk 1 above)
Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/TableCache.java]: Ok. Logic moved.

File: plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/meta/GenericMetaModel.java [hunk 2]
```diff
@@ -724,6 +725,21 @@ public GenericTableColumn createTableColumnImpl(@NotNull DBRProgressMonitor moni
...
 
+    /**
+     * Will set precision for type from length if type can have precision
+     *
+     * @param valueType type id
+     * @param columnSize length of the column
+     * @return precision of the numeric column or null
+     */
+    @Nullable
+    public Integer setPrecisionForNumericColumnTypes(int valueType, long columnSize) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/meta/GenericMetaModel.java]: Why set? maybe extractPrecisionOfNumericColumn

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/meta/GenericMetaModel.java (hunk 2 above)
Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/meta/GenericMetaModel.java]: Ok

**Comment by uslss** [comment]
verified

**Review by serge-rider — APPROVED** [review]

**Pull request merged**